src/main/java/org/apache/brooklyn/cloudfoundry/location/CloudFoundryPaasLocation.java

```diff
@@ -387,4 +408,224 @@ public int getMemory(String applicationName) {
+                    .block(getConfig(OPERATIONS_TIMEOUT));
+        } catch (Exception e) {
+            log.error("Error gettin the service {} the error was {}", serviceInstanceName, e);
+            throw new PropagatedRuntimeException(e);
+        }
+    }
+
+    public void deleteServiceInstance(String serviceInstanceId) {
+        try {
+            getClient().services()
+                    .deleteInstance(DeleteServiceInstanceRequest.builder()
+                            .name(serviceInstanceId)
+                            .build())
+                    .doOnSuccess(v -> log.info("Deleted service instance {}", serviceInstanceId))
+                    .block(getConfig(OPERATIONS_TIMEOUT));
+        } catch (Exception e) {
+            log.error("Error deleting service {}, the error was {}", serviceInstanceId, e);
+            throw new PropagatedRuntimeException(e);
+        }
+    }
+
+    public void bindServiceToApplication(String serviceInstanceName, String applicationName) {
+        try {
+            getClient().services()
+                    .bind(BindServiceInstanceRequest.builder()
+                            .applicationName(applicationName)
+                            .serviceInstanceName(serviceInstanceName)
+                            .build())
+                    .doOnSuccess(v -> log.info("Bound service instance {} to application {}",
+                            serviceInstanceName, applicationName))
+                    .block(getConfig(OPERATIONS_TIMEOUT));
+        } catch (Exception e) {
+            log.error("Error binding the service {} to the application {}, the error was {}",
+                    new Object[]{serviceInstanceName, applicationName, e});
+            throw new PropagatedRuntimeException(e);
+        }
+    }
+
+    public void unbindService(String serviceName, String applicationName) {
+        try {
+            getClient().services()
+                    .unbind(UnbindServiceInstanceRequest.builder()
+                            .applicationName(applicationName)
+                            .serviceInstanceName(serviceName)
+                            .build())
+                    .doOnSuccess(v -> log.info("Unbound service instance {} to application {}",
+                            serviceName, applicationName))
+                    .block(getConfig(OPERATIONS_TIMEOUT));
+        } catch (Exception e) {
+            log.error("Error unbinding service {} to application {} the error was {}",
+                    new Object[]{serviceName, applicationName, e});
+            throw new PropagatedRuntimeException(e);
+        }
+    }
+
+    public boolean isServiceBoundTo(String serviceName, String applicationName) {
+        return getServiceInstance(serviceName).getApplications().contains(applicationName);
+    }
+
+    public List<String> getBoundApplications(String serviceInstanceName) {
+        return getServiceInstance(serviceInstanceName).getApplications();
```

Comment:
in this case, if the serviceInstanceName does not exist and exception will be thrown, does not it make sense?